src/psyclone/psyir/tools/dependency_tools.py

```diff
@@ -156,6 +156,8 @@ def __init__(self, loop_types_to_parallelise=None):
         else:
             self._loop_types_to_parallelise = []
         self._clear_messages()
+        # Cache to hold most recently generated VariablesAccessInfo.
+        self._var_access_info = None
```

Comment:
If you end up using the self.reference_access in the Directive update you could revert this changes. In general having to access the results from a cache after the execution is not an ideal API, and is easy to inadvertently call something in the middle that changes the results.

Reply:
Reverted.